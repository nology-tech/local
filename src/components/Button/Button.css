@import url("https://fonts.googleapis.com/css2?family=Inter&family=Lato&family=Roboto&display=swap");
* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

/*                          Colours                         */
/*                          Font stuff              */
* {
  font-family: "Trebuchet MS", "Lucida Sans Unicode", "Lucida Grande", "Lucida Sans", Arial, sans-serif;
  color: #000000;
}

h1,
h2,
h3,
h4,
h5,
h6,
p {
  margin: 0.5rem 0;
  display: inline-block;
  width: 100%;
}

button {
  font-size: 1rem;
}

h1 {
  font-size: 2.1rem;
}

h2 {
  font-size: 1.65rem;
}

h3 {
  font-size: 1.25rem;
}

@media screen and (min-width: 768px) {
  :root {
    font-size: 110%;
  }
}
@media screen and (min-width: 1024px) {
  :root {
    font-size: 120%;
  }
}
.button {
  font-family: "Inter";
  background-color: #1f1b1b;
  width: 9rem;
  height: 3rem;
  font-size: 1rem;
  border: 0.1rem solid black;
  opacity: 0.8;
}
.button:hover {
  opacity: 1;
  transition: 0.3s;
}
.button:active {
  border: 0.2rem solid #f3f2e8;
  opacity: 0.4;
}

.button--contact {
  height: 4rem;
  margin: auto;
  width: 12rem;
  color: #f3f2e8;
  background-color: #5bb27c;
  border-color: #5bb27c;
  font-size: 3rem;
  font-weight: 400;
  border-radius: 1.7rem;
}/*# sourceMappingURL=Button.css.map */