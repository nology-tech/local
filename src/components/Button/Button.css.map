{"version":3,"sources":["../../styles/_typography.scss","../../styles/_reset.scss","Button.css","../../styles/_variables.scss","Button.scss"],"names":[],"mappings":"AACQ,oGAAA;ACDR;EACE,UAAA;EACA,SAAA;EACA,sBAAA;ACEF;;ACEA,6DAAA;AAuBA,qDAAA;AHxBA;EACE,qGAAA;EAEA,cGCW;ADEb;;AFAA;;;;;;;EAQE,gBAAA;EACA,qBAAA;EACA,WAAA;AEEF;;AFCA;EACE,eAAA;AEEF;;AFCA;EACE,iBAAA;AEEF;;AFCA;EACE,kBAAA;AEEF;;AFCA;EACE,kBAAA;AEEF;;AFCA;EAEE;IACE,eAAA;EECF;AACF;AFEA;EAEE;IACE,eAAA;EEDF;AACF;AE9CA;EACE,oBD6BiB;EC5BjB,yBDIiB;ECFjB,WAAA;EAEA,YAAA;EAEA,eD+BkB;EC7BlB,0BAAA;EACA,YAAA;AF4CF;AE3CE;EACE,UAAA;EACA,gBAAA;AF6CJ;AE3CE;EACE,4BAAA;EACA,YAAA;AF6CJ;;AEzCA;EACE,YAAA;EACA,YAAA;EACA,YAAA;EACA,cDVM;ECWN,yBDRoB;ECSpB,qBDToB;ECUpB,eAAA;EAEA,gBDIoB;ECFpB,qBAAA;AF0CF","file":"Button.css"}