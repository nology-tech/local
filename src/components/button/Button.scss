@use "../../../src/styles/reset" as *;
@use "../../../src/styles/variables" as *;
@use "../../styles/typography" as *;
@use "../../styles/mixins" as *;

.button {
  background-color: $background-color;

  width: 9rem;

  height: 3rem;

  font-size: $default-text-size;

  padding: 10px 20px;

  border: solid black;
}

.button-contact {
  margin: auto;
  width: 12rem;
  font-family: $text-Font-Family;
  color: $white;
  margin-top: 10px;
  padding: 0.5rem 0.9rem;

  background-color: $green-color-palette;

  font-size: 3rem;

  font-weight: $default-font-weight;

  border-radius: 25px;

  border: none;
}
