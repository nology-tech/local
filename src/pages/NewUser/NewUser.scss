@use "../../styles/variables" as *;
.newuser {
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100%;

  &__description {
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    align-items: center;
    text-align: center;
    font-size: 1.3rem;
    height: 50%;
    width: 80%;
    &--content {
      position: relative;
      top: 5rem;
    }
  }
  &__logo {
    width: 50%;
  }

  &__buttons {
    height: 50%;
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    position: relative;
    top: 2.5rem;
  }
}
