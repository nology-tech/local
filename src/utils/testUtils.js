import { jest } from "@jest/globals";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import UserContext from "../context/UserContext";
import mockUser from "../data/mockUser";

export const customRender = (
  ui,
  useRouting = true,
  contextOptions = {
    useContext: true,
    value: { user: mockUser, setUser: jest.fn() },
  }
) => {
  // wrap components in routing and context if requested
  const routingUiResult = useRouting && wrapWithRouting(ui);
  const contextUiResult =
    contextOptions.useContext &&
    wrapWithContext(routingUiResult, contextOptions.value);

  // use RTL's render function to return the test component
  return render(contextUiResult);
};

const wrapWithRouting = (ui) => {
  return <Router>{ui}</Router>;
};

const wrapWithContext = (ui, value) => {
  return <UserContext.Provider value={value}>{ui}</UserContext.Provider>;
};
