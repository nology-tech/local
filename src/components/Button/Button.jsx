import "./Button.scss";

const Button = ({
  buttonText,
  buttonName,
  onClick,
  // buttonImgUrl,
  // buttonImgAlt,
  // buttonImgClass,
}) => {
  let buttonClass = `button button--${buttonName}`;
  // let buttonImageName = `button__${buttonImgClass}`;
  // let buttonImageUrl = `${buttonImgUrl}`;

  // const buttonImage = () => {
  //   if (typeof buttonImageUrl == "defined") {
  //     return (
  //       <img
  //         className={buttonImageName}
  //         src={buttonImageUrl}
  //         alt={buttonImgAlt}
  //       />
  //     );
  //   }
  // };

  return (
    <button className={buttonClass} onClick={onClick}>
      {buttonText}
      {/* {buttonImage} */}
    </button>
  );
};

export default Button;
