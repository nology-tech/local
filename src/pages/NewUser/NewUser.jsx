import Layout from "../../components/Layout/Layout";
import Button from "../../components/Button/Button";
import "./NewUser.scss";
import logo from "../../assets/images/AppLogo.svg";
import { useState } from "react";
import { Link } from "react-router-dom";

const NewUser = () => {
  const [buttonText, setButtonText] = useState("Get Started");
  const [content, setContent] = useState(
    " Bringing local business to local people"
  );

  const buttonClicked = () => {
    if (buttonText == "Get Started") {
      setButtonText("Continue");
      setContent(
        "No matter where you are in the world Local shows you the best local bars, cafes, restaurants and shops"
      );
    } else if (buttonText == "Continue") {
      setButtonText("Sign Up");
      setContent(
        "Discover new places and curate your own personal list. Be a local anywhere"
      );
    }
  };

  const buttonJSX = (
    <Button
      onClick={buttonClicked}
      buttonText={buttonText}
      buttonName="enter"
    />
  );

  const signUpButtonJSX = (
    <>
      <Link to="/enter/signin">
        <Button
          onClick={buttonClicked}
          buttonText="Sign In"
          buttonName="enter"
        />
      </Link>
      <Link to="/enter/signup">
        <Button
          onClick={buttonClicked}
          buttonText={buttonText}
          buttonName="enter"
        />
      </Link>
    </>
  );

  return (
    <Layout>
      <div className="newuser">
        <div className="newuser__description">
          <div className="newuser__header">
            <img className="newuser__logo" src={logo} alt="App Logo" />
            <h1>LOCAL</h1>
          </div>

          <p className="newuser__description--content">{content}</p>
        </div>
        <div className="newuser__buttons">
          {buttonText == "Sign Up" ? signUpButtonJSX : buttonJSX}
        </div>
      </div>
    </Layout>
  );
};

export default NewUser;
