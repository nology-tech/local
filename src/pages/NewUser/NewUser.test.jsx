import NewUser from "./NewUser";
import { customRender } from "../../utils/testUtils";
import Button from "../../components/Button/Button";
import { screen } from "@testing-library/react";

import userEvent from "@testing-library/user-event";
it("should render NewUser page", () => {
  const { container } = customRender(<NewUser />);

  expect(container).toMatchSnapshot();
});

it("should render the content of the page and the button when user clicks", () => {
  const mockFunction = jest.fn();

  customRender(<Button onClick={mockFunction} />);

  const button = screen.getByRole("button");
  userEvent.click(button);

  expect(mockFunction).toBeCalled();
});
