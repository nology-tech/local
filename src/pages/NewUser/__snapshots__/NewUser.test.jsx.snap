// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NewUser page 1`] = `
<div>
  <div
    class="layout"
  >
    <div
      class="newuser"
    >
      <div
        class="newuser__description"
      >
        <div
          class="newuser__header"
        >
          <img
            alt="App Logo"
            class="newuser__logo"
            src="AppLogo.svg"
          />
          <h1>
            LOCAL
          </h1>
        </div>
        <p
          class="newuser__description--content"
        >
           Bringing local business to local people
        </p>
      </div>
      <div
        class="newuser__buttons"
      >
        <button
          class="button button--enter"
        >
          Get Started
        </button>
      </div>
    </div>
  </div>
</div>
`;
